### PR TITLE
Create a generic one roster class that infinitecampus inherits from

### PR DESCRIPTION
**Pull request by JohnnyPane**
Skyward/OneRoster was having issues connecting with the latest changes. Now we're connecting directly to OneRoster. Infinite campus now just inherits from one roster. All OneRoster related systems will be namespaced under the OneRoster class

Moved some specific filtering in to InfiniteCampus, as far as I can tell one roster uses userIds for all responses

I tested just about everything here but really dug in to get_student/get_student. Find a skyward district and create an instance of Bright::SisApi::OneRoster::Skyward instead of Bright::SisApi::Skyward. I also use a district with infinite campus to compare results with. 

When these changes are made we also need to [make changes](https://github.com/Arux-Software/switchboard_sis_api/pull/64) to the switchboard_sis_api

### CODE REVIEW

File: lib/bright/sis_apis/infinite_campus.rb [hunk 1]
```diff
@@ -1,387 +1,17 @@
-require 'oauth'
+require_relative 'one_roster'
```
Comment (brgeen) [inline lib/bright/sis_apis/infinite_campus.rb:1]: This can be removed and then added to the `bright.rb` manifest file

**Review by brgeen — COMMENTED** [review]
Do we want to make this architecture change to the other SIS classes?
```
- Base
    - OneRoster
        - Aeries
        - Focus
        - InfiniteCampus
        - PowerSchools
        - Skyward
        - Synergy
        - Tsis
```
Any duplicate logic can be put into the parent class, either `OneRoster` or `Base`. 

**Thinking long-term here:** 
If most of the SIS providers are using oneroster already, we could switch them over to the `OneRoster` connection like @JohnnyPane mentioned.

Then we could create an audit of districts that should be migrated to the generic OneRoster class, and as we do this, we could remove the domain specific classes in here, and ideally just have a `OneRoster` class

File: lib/bright/sis_apis/infinite_campus.rb [hunk 2]
```diff
@@ -1,387 +1,15 @@
...
   module SisApi
-    class InfiniteCampus < Base
-
-      @@description = "Connects to the Infinite Campus OneRoster API for accessing student information"
-      @@doc_url = "https://content.infinitecampus.com/sis/latest/documentation/oneroster-api"
-      @@api_version = "1.1"
-
-      attr_accessor :connection_options, :schools_cache, :school_years_cache
-
-      DEMOGRAPHICS_CONVERSION = {
```
[the remark this answers is not in the document]
Comment (snovotny) [reply lib/bright/sis_apis/infinite_campus.rb:13]: Although that might not be a problem when it's all on OneRoster... ?

[the remark this answers is not in the document]
Comment (JohnnyPane) [reply lib/bright/sis_apis/infinite_campus.rb:13]: Specifically the DEMOGRAPHICS_CONVERSION? I've been doing some digging and it looks like the one roster api returns demographic info in the same way. So I think it looks good?

File: lib/bright/sis_apis/one_roster.rb [hunk 3]
```diff
@@ -0,0 +1,385 @@
+require 'oauth'
+
+module Bright
+  module SisApi
+    class OneRoster < Base
+      @@description = "Connects to the Infinite Campus OneRoster API for accessing student information"
+      @@doc_url = "https://content.infinitecampus.com/sis/latest/documentation/oneroster-api"
+      @@api_version = "1.1"
```
[the remark this answers is not in the document]
Comment (JohnnyPane) [reply lib/bright/sis_apis/one_roster.rb]: Updated in the latest commit, will have to add api_version attributes to districts sis_settings for districts that are using the old uri/secret/client settings

File: lib/bright/sis_apis/infinite_campus.rb [hunk 4]
```diff
@@ -1,387 +1,17 @@
-require 'oauth'
+require_relative "one_roster"
```
Comment (snovotny) [inline lib/bright/sis_apis/infinite_campus.rb:1]: Does the `skyward` one need this?

**Comment by snovotny** [comment]
Let's bump the version too I think (1.3):
https://github.com/Arux-Software/Bright/blob/b07571c7936e4653bad3e945bf5304d319484294/lib/bright/version.rb#L2

File: lib/bright/sis_apis/infinite_campus.rb (hunk 4 above)
Comment (JohnnyPane) [reply lib/bright/sis_apis/infinite_campus.rb:1]: From what I can tell skyward doesn't need this but the issue is I'm not sure why one would and the other wouldn't...

**Review by sheuer — APPROVED** [review]

File: lib/bright/student.rb [hunk 5]
```diff
@@ -19,27 +19,26 @@ def self.attribute_names
...
+      "#{first_name} #{middle_name} #{last_name}".gsub(/\s+/, " ").strip
     end
 
     def <=>(other)
-      (self.sis_student_id and self.sis_student_id == other.sis_student_id) or
-      (self.state_student_id and self.state_student_id == other.state_student_id) or
-      (self.first_name == other.first_name and self.middle_name == other.middle_name and self.last_name == other.last_name and self.birth_date == other.birth_date)
+      (sis_student_id and sis_student_id == other.sis_student_id) or
+        (state_student_id and state_student_id == other.state_student_id) or
+        (first_name == other.first_name and middle_name == other.middle_name and last_name == other.last_name and birth_date == other.birth_date)
```
Comment (chriswarren) [inline lib/bright/student.rb:31]:
```suggestion
      (sis_student_id && sis_student_id == other.sis_student_id) ||
        (state_student_id && state_student_id == other.state_student_id) ||
        (first_name == other.first_name && middle_name == other.middle_name && last_name == other.last_name && birth_date == other.birth_date)
```

**Review by chriswarren — APPROVED** [review]

**Pull request merged**